ambry-api/src/main/java/com.github.ambry/rest/RestRequestMetricsTracker.java

```diff
@@ -237,6 +237,25 @@ public void recordMetrics() {
     }
   }
 
+  public long getRoundTripTimeInMs() {
+    return nioMetricsTracker.roundTripTimeInMs;
+  }
+
+  public long getTimeToFirstByteInMs(){
+    return nioMetricsTracker.timeToFirstByteInMs;
+  }
+
+  public void updateUnSatisfiedRequestCounter(){
```

Comment:
I noticed that the metrics would be injected later and `recordMetrics` is called when closing the request. Hence, the `updateUnsatisfiedRequestCounter` is called only after request is closed. I see your point here, maybe I should try to integrate it with `recordMetrics` in case anyone invoke this in other places.